### PR TITLE
Release v0.7 disable gRPC on non supported platforms

### PR DESCRIPTION
**Pull request by vincentpierre**
This would be a quick fix for disabling training on unsupported platforms.
Might still need to document that if the user wants to recompile the protobuf objects, they will need to edit the generated files.

I tested it on UWP

### CODE REVIEW

File: UnitySDK/Assets/ML-Agents/Scripts/CommunicatorObjects/UnityToExternalGrpc.cs [hunk 1]
```diff
@@ -2,6 +2,9 @@
 //     Generated by the protocol buffer compiler.  DO NOT EDIT!
 //     source: mlagents/envs/communicator_objects/unity_to_external.proto
 // </auto-generated>
+
+# if UNITY_EDITOR ||  UNITY_STANDALONE_WINDOWSWINDOWS || UNITY_STANDALONE_OSX || UNITY_STANDALONE_LINUX
```
Comment (mantasp) [inline UnitySDK/Assets/ML-Agents/Scripts/CommunicatorObjects/UnityToExternalGrpc.cs]: UNITY_STANDALONE_WINDOWSWINDOWS looks like a typo

Comment (vincentpierre) [reply UnitySDK/Assets/ML-Agents/Scripts/CommunicatorObjects/UnityToExternalGrpc.cs]: Will fix

File: protobuf-definitions/README.md [hunk 2]
```diff
@@ -26,3 +26,14 @@ On Windows & Linux: [See here](https://github.com/google/protobuf/blob/master/sr
 1. Install pre-requisites.
 2. Un-comment line 4 in `make.bat`, and set to correct Grpc.Tools sub-directory.
 3. Run `make.bat`
+4. In the generated `UnityToExternalGrpc.cs` file in the `CommunicatorObjects` folder, you will need to add on top of the file the line
```
Comment (awjuliani) [inline protobuf-definitions/README.md]: Say where `CommunicatorObjects` folder is (i.e. in UnitySDK). 

Also change  "to add on top of the file the line"
to
"to add the following to the beginning of the file"

File: protobuf-definitions/README.md [hunk 3]
```diff
@@ -26,3 +26,14 @@ On Windows & Linux: [See here](https://github.com/google/protobuf/blob/master/sr
...
+
+```csharp
+# if UNITY_EDITOR || UNITY_STANDALONE_WIN || UNITY_STANDALONE_OSX || UNITY_STANDALONE_LINUX
+```
+ and at the bottom of the file the line
```
Comment (awjuliani) [inline protobuf-definitions/README.md]: "and the following line to the end"

File: protobuf-definitions/README.md [hunk 4]
```diff
@@ -26,3 +26,14 @@ On Windows & Linux: [See here](https://github.com/google/protobuf/blob/master/sr
...
+
+```csharp
+# if UNITY_EDITOR || UNITY_STANDALONE_WIN || UNITY_STANDALONE_OSX || UNITY_STANDALONE_LINUX
+```
+ and at the bottom of the file the line 
+ 
+ ```csharp
+ #endif
+ ```
+This is to make sure the Grpc library is not used on platforms that do not support training.
```
Comment (awjuliani) [inline protobuf-definitions/README.md]: Explain why this is needed.

Comment (vincentpierre) [reply protobuf-definitions/README.md]: You mean more than the current explanation ? What is missing in it ? I say It is to make sure the Grpc library is not used on platforms that do not support training

**Pull request merged**